### PR TITLE
feat: extended integration test

### PR DESCRIPTION
**Pull request by Polybius93**

### CODE REVIEW

File: it/src/index.js [hunk 1]
```diff
@@ -108,9 +107,44 @@ async function waitForBalance(dlcManager) {
...
+  if (balanceAfterFunding + collateralAmount !== balanceAfterClosing) {
+    console.log('Error: Balance after closing does not match the expected value');
+  } else {
+    console.log('Balance after closing matches the expected value');
+  }
+}
+
+async function fetchTxDetails(txId) {
+  try {
+    const res = await fetch(`https://devnet.dlc.link/electrs/tx/${txId}`, {
```
Comment (sosaucily) [inline it/src/index.js:106]: This is already an ENV var you can use

File: it/src/index.js [hunk 2]
```diff
@@ -108,9 +107,44 @@ async function waitForBalance(dlcManager) {
...
+async function checkBalance(dlcManager, action, timeout) {
+  let remainingTime = timeout;
+  while (remainingTime > 0) {
+    console.log(`Checking balance in ${remainingTime / 1000} seconds...`);
+    await new Promise((resolve) => setTimeout(resolve, 10000));
+    remainingTime -= 10000;
+  }
+  const balance = await dlcManager.get_wallet_balance();
+  console.log(`DLC Wasm Wallet Balance at ${action}: ` + balance);
+  return balance;
```
Comment (sosaucily) [inline it/src/index.js]: This feels a little over complicated for doing something pretty simple, let's see if we can simplify

Comment (Polybius93) [reply it/src/index.js]: This is outdated!

File: it/src/index.js [hunk 3]
```diff
@@ -152,66 +172,100 @@ async function main() {
...
+  }
+
+  //Check if the contract is in the Closed state
+  routerWalletInfo = await (await fetch(`${protocolWalletURL}/info`)).json();
+  checkIfContractIsInState(contractID, routerWalletInfo.contracts, 'Closed');
+
+  //Check if the balance increased after closing the contract
+  checkBalance(dlcManager, '[CONTRACT CLOSED]').then((balanceAfterClosing) =>
+    checkBalanceAfterClosing(Number(balanceAfterFunding), Number(balanceAfterClosing), Number(acceptCollateral))
+  );
```
Comment (sosaucily) [inline it/src/index.js:268]:
```js
assert(x + y === z, "expected the thing to be right brah");

const assert = (predicate: bool, message: String) => if (!predicate) {
print message;
sys.exit(1);
}
```

File: it/src/index.js [hunk 4]
```diff
@@ -114,10 +102,29 @@ async function fetchTxDetails(txId) {
   }
 }
 
+async function retry(checkFunction, timeoutTime) {
+  let timeRemaining = timeoutTime;
+  while (timeRemaining) {
+    const result = await checkFunction();
+    if (result) return true;
+    await new Promise((resolve) => setTimeout(resolve, 10000));
```
Comment (sosaucily) [inline it/src/index.js:110]: Change this to 1 or 2 seconds, or maybe 5. No need to wait for 10 to do a check I'd say. If 2 seconds doesn't harm anything use that. The point of having this was that we would retry often to speed things up

File: it/src/index.js [hunk 5]
```diff
@@ -247,25 +251,31 @@ async function main() {
...
   const confirmedClosingTransaction = await waitForConfirmations(blockchainHeightAtBroadcast, 6);
   if (confirmedClosingTransaction) {
     console.log('Closing transaction confirmed');
   }
 
   //Check if the contract is in the Closed state
-  routerWalletInfo = await (await fetch(`${protocolWalletURL}/info`)).json();
-  checkIfContractIsInState(contractID, routerWalletInfo.contracts, 'Closed');
+  assert(
+    retry(() => checkIfContractIsInState(contractID, 'Closed'), 15000),
```
Comment (sosaucily) [inline it/src/index.js:268]: nice, this looks good 👍

File: it/src/index.js [hunk 6]
```diff
@@ -247,25 +251,31 @@ async function main() {
...
+    `Contract state is not updated in the Router Wallet to Closed`
+  );
 
   //Check if the balance increased after closing the contract
-  checkBalance(dlcManager, '[CONTRACT CLOSED]').then((balanceAfterClosing) =>
-    checkBalanceAfterClosing(Number(balanceAfterFunding), Number(balanceAfterClosing), Number(acceptCollateral))
-  );
+  checkBalance(dlcManager, '[CONTRACT CLOSED]').then((balanceAfterClosing) => {
+    assert(
```
Comment (sosaucily) [inline it/src/index.js:274]: yeah i think this is better 👍 thanks

File: it/src/index.js [hunk 7]
```diff
@@ -126,21 +133,17 @@ async function waitForConfirmations(blockchainHeightAtBroadcast, targetConfirmat
   return true;
 }
 
-function checkIfContractIsInState(contractID, contracts, state) {
-  if (contracts[state].includes(contractID)) {
-    console.log(`Contract state is updated in the Router Wallet to ${state}`);
-  } else {
-    console.error(`Contract state is not updated in the Router Wallet to ${state}`);
-    process.exit(1);
+async function checkIfContractIsInState(contractID, state) {
```
Comment (sosaucily) [inline it/src/index.js:136]: nice to have this function, as you use this logic in lots of places

File: it/src/index.js [hunk 8]
```diff
@@ -126,21 +133,17 @@ async function waitForConfirmations(blockchainHeightAtBroadcast, targetConfirmat
...
+  const routerWalletInfo = await (await fetch(`${protocolWalletURL}/info`)).json();
+  if (routerWalletInfo.contracts[state].includes(contractID)) {
+    return true;
```
Comment (sosaucily) [inline it/src/index.js:139]: optimize this to 

```js
return routerWalletInfo.contracts[state].includes(contractID);
```

**Review by sosaucily — APPROVED** [review]

**Pull request merged**